gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineConfig.java

```diff
@@ -37,23 +37,33 @@ public void apply(Project rootProject) {
 
         // users can still override this default dependency, it just reduces boilerplate
         Optional<String> version = Optional.ofNullable(getClass().getPackage().getImplementationVersion());
-        configuration.defaultDependencies(d -> d.add(project.getDependencies().create(String.format(
+        configuration.defaultDependencies(d -> d.add(rootProject.getDependencies().create(String.format(
                 "com.palantir.baseline:gradle-baseline-java-config%s@zip", version.map(v -> ":" + v).orElse("")))));
 
         // Create task for generating configuration.
-        rootProject.getTasks().create("baselineUpdateConfig", task -> {
+        rootProject.getTasks().register("baselineUpdateConfig", task -> {
             task.setGroup("Baseline");
             task.setDescription("Installs or updates Baseline configuration files in .baseline/");
+            task.getInputs().files(configuration);
+            task.getOutputs().dir(getConfigDir());
+            task.getOutputs().dir(rootProject.getRootDir().toPath().resolve("project"));
             task.doLast(t -> {
-
                 if (configuration.getFiles().size() != 1) {
                     throw new IllegalArgumentException("Expected to find exactly one config dependency in the "
                             + "'baseline' configuration, found: " + configuration.getFiles());
                 }
 
                 rootProject.copy(copySpec -> {
-                    copySpec.from(project.zipTree(configuration.getSingleFile()));
+                    copySpec.from(rootProject.zipTree(configuration.getSingleFile()));
                     copySpec.into(getConfigDir());
+                    copySpec.exclude("**/scalastyle_config.xml");
+                    copySpec.setIncludeEmptyDirs(false);
+                });
+                rootProject.copy(copySpec -> {
+                    copySpec.from(rootProject.zipTree(configuration.getSingleFile())
+                            .filter(file -> file.getName().equals("scalastyle_config.xml")));
+                    copySpec.into(rootProject.getRootDir().toPath().resolve("project"));
```

Comment:
Let's put a tiny comment to explain why we're putting stuff into "$rootProject/project/scalastyle_config.xml"

Reply:
does this mean that I'll have a new directory, `project` in every repo from now on? not a huge fan of that churn, given we have only a very small amount of scala repos.

Reply:
I pushed a fix for this - the file is only created if there exists a project that has scala plugin applied to it. Should also make sure that scala style is applied as well